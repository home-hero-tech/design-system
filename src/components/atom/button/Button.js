import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import css from './Button.module.scss';

/* <Button type="primary" container="outlined" shape="squared" size="" /> */

const Button = ({ submit, shape, size, container, type, children }) => {
  const btnType = submit ? 'submit' : 'button';

  const classes = classNames({
    [css['c-btn']]: true,
    [css[`c-btn--${type}`]]: true,
    [css[`c-btn--${shape}`]]: true,
    [css[`c-btn--${container}`]]: true,
    [css[`c-btn--size-${size}`]]: true
  });

  return (
    <button type={btnType} className={classes}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.string.isRequired,
  type: PropTypes.oneOf([
    'default',
    'primary',
    'secondary',
    'danger',
    'success'
  ]),
  submit: PropTypes.bool,
  size: PropTypes.oneOf([1, 2, 3, 4]),
  container: PropTypes.oneOf(['default', 'outlined', 'text']),
  shape: PropTypes.oneOf(['squared', 'rounded'])
};

Button.defaultProps = {
  submit: false,
  type: 'default',
  container: 'default',
  shape: 'rounded',
  size: 2
};

export default Button;
