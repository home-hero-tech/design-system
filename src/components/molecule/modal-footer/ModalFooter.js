import React from 'react';
import PropTypes from 'prop-types';

import css from '../../organism/modal/Modal.module.scss';

const ModalFooter = ({ children }) => (
  <footer className={css['c-modal__actions']}>{children}</footer>
);

ModalFooter.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired
};

ModalFooter.defaultProps = {};

ModalFooter.displayName = 'ModalFooter';

export default ModalFooter;
