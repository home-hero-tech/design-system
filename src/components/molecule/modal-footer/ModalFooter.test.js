import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ModalFooter from './ModalFooter';
import Button from '../../atom/button/Button';

describe('ModalFooter', () => {
  it('should match with the snapshot', () => {
    const children = (
      <>
        <Button onClick={() => {}}>1st action</Button>
        <Button onClick={() => {}} type="primary">
          2nd action
        </Button>
      </>
    );
    const wrapper = shallow(<ModalFooter>{children}</ModalFooter>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
