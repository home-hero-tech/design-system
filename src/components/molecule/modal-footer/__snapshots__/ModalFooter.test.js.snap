// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalFooter should match with the snapshot 1`] = `
<footer
  className="c-modal__actions"
>
  <Button
    container="default"
    disabled={false}
    loading={false}
    onClick={[Function]}
    shape="rounded"
    size={2}
    submit={false}
    type="default"
  >
    1st action
  </Button>
  <Button
    container="default"
    disabled={false}
    loading={false}
    onClick={[Function]}
    shape="rounded"
    size={2}
    submit={false}
    type="primary"
  >
    2nd action
  </Button>
</footer>
`;
